nodes/src/nodes/library/ssrf_protection.py

```
@@ -0,0 +1,282 @@
+    Raises
+    ------
+    SSRFError
+        If the URL targets a blocked IP, uses a disallowed scheme, or
+        cannot be resolved.
+    """
+    parsed = urlparse(url)
+
+    # -- Scheme check -------------------------------------------------------
+    scheme = (parsed.scheme or '').lower()
+    if scheme not in _ALLOWED_SCHEMES:
+        raise SSRFError(f'SSRF protection: scheme {scheme!r} is not allowed. Only {sorted(_ALLOWED_SCHEMES)} are permitted.')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)